packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -256,7 +272,7 @@ export async function getPageStaticInfo(params: {
 
   const fileContent = (await tryToReadFile(pageFilePath, !isDev)) || ''
   if (
-    /runtime|getStaticProps|getServerSideProps|matcher|unstable_allowDynamic/.test(
+    /runtime|getStaticProps|getServerSideProps|matcher|unstable_allowDynamic|export (let|var|const) config/.test(
```

Comment:
**sugestion:** replace with `/runtime|getStaticProps|getServerSideProps|export const config/`

The reason why we had `matcher` and `unstable_allowDynamic` was to catch files with user config. But your idea of checking directly the presence of `export const config` is better than checking individual property names.

We can ignore `export let config` and `export var config` because the function `extractExportedConstValue()` ([here](https://github.com/vercel/next.js/blob/canary/packages/next/build/analysis/extract-const-value.ts#L219)) ignores them on purpose.

I remember discussing this with @javivelasco, and we agreed that it is preferable to prevent users having a non-const exported key.

Reply:
<picture data-single-emoji=":sgtm:" title=":sgtm:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..1kiwROaloy6zLYQj.34M-VA-BMO28KtMnTj50k2SgYkDRdqmEDVRkYCrLLVDLJLP_aZgJcnM7AEId3oV_u47nc5Q2H1IsOQ9XArap1jkKqnoCInMIkvYrX38g1vPcgAtcXNU.7U3WRmsd7tG0MPCiE5JVBw" alt=":sgtm:" align="absmiddle"></picture> 
will do that tomorrow